src/group/managed_group/errors.rs

```diff
@@ -2,74 +2,54 @@
 //!
 //! `WelcomeError`, `ApplyCommitError`, `DecryptionError`, and
 //! `CreateCommitError`.
-//!
-//! #[derive(PartialEq, Debug)]
 
 use crate::codec::CodecError;
 use crate::config::ConfigError;
+use crate::error::{ErrorPayload, ErrorString};
 use crate::group::{ApplyCommitError, CreateCommitError, ExporterError, GroupError};
 
-use std::error::Error;
-
-#[derive(PartialEq, Debug)]
-pub enum ManagedGroupError {
-    Unknown,
-    Codec(CodecError),
-    Config(ConfigError),
-    Group(GroupError),
-    CreateCommit(CreateCommitError),
-    UseAfterEviction,
-    PendingProposalsExist,
-    Exporter(ExporterError),
-}
-
-impl From<ConfigError> for ManagedGroupError {
-    fn from(err: ConfigError) -> ManagedGroupError {
-        ManagedGroupError::Config(err)
-    }
-}
-
-impl From<CodecError> for ManagedGroupError {
-    fn from(err: CodecError) -> ManagedGroupError {
-        ManagedGroupError::Codec(err)
+implement_error! {
+    pub enum ManagedGroupError {
+        LibraryError(ErrorString) =
```

Comment:
We should define some rules on when we want to return a `LibraryError`. We don't have to get rid of all `.unwrap()`s if can clearly document that it is safe, but we should define where the threshold is.

Reply:
Yeah this is a generally contagious issue. I'm generally in favor of either using errors or not using errors. Since we want to use errors I prefer always using them. `unwrap` should only be used when there's obviously no way that the error could happen. But this is rarely the case. In any other case we should be using errors. With proper handling like I started here, this should never be more than a `?`, which is actually more readable in code than `unwrap()`.

Reply:
I filed #261 for that, I think this is good enough as a starting point.